### PR TITLE
Feat: Add a single endpoint to fetch community page data

### PR DESCRIPTION
**Pull request by abdullai-t**
####  Summary / Highlights
This pull request includes:

- A new API endpoint designed to retrieve necessary data for rendering a specific page on the community portal. This endpoint will streamline data fetching, ensuring that all relevant information for the page is gathered in a single request, thereby improving efficiency and load times.
- Updates the community portal's menu items endpoint to include necessary computations for determining active items. Additionally, the appropriate subdomain is now prepended to each menu item link. 

#### Requirements (place an `x` in each `[ ]`)
* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [x] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

### CODE REVIEW

File: src/api/services/misc.py [hunk 1]
```diff
@@ -141,3 +141,20 @@ def authenticateFrontendInTestMode(self, args):
             return None, CustomMassenergizeError(str(err))
         client = Client()
         return signinAs(client, user), None
+    
+    
+    def load_essential_initial_site_data(self,context, args):
```
Comment (BradHN1) [inline src/api/services/misc.py:146]: Is this used?  If not should remove

File: src/api/store/misc.py [hunk 2]
```diff
@@ -492,3 +494,83 @@ def list_commonly_used_icons(self):
         sorted_keys = sorted(common_icons, key=common_icons.get, reverse=True)
         for key in sorted_keys:
             print(str(key) + ": " + str(common_icons[key]))
+            
+    
+    def load_essential_initial_site_data(self, context, args):
```
Comment (BradHN1) [inline src/api/store/misc.py:499]: Is this used?  If not should remove

File: src/api/utils/api_utils.py [hunk 3]
```diff
@@ -98,3 +108,283 @@ def create_media_file(file, name):
...
+                    item["children"].remove(child)
+        
+    for item in menu_items:
+        f = prependPrefixToLinks(item,prefix)
+        main_menu.append(f)
+
+    return main_menu
+
+def get_viable_menu_items(community):
+
```
Comment (BradHN1) [inline src/api/utils/api_utils.py:151]: why isn't DonatePageSettings included in this?

File: src/api/utils/api_utils.py [hunk 4]
```diff
@@ -98,3 +108,283 @@ def create_media_file(file, name):
...
+            vendors = vendors.filter(Q(user__id=context.user_id) | Q(is_published=True))
+        else:
+            vendors = vendors.filter(is_published=True)
+    return vendors.distinct()
+
+def get_tags_collections():
+    return TagCollection.objects.filter(is_deleted=False)
+
+# .......................... site setup utils ..........................
+def load_homepage_data(context, args, community, id, page_settings):
```
Comment (BradHN1) [inline src/api/utils/api_utils.py:264]: These methods aren't used?  If not remove

Comment (abdullai-t) [reply src/api/utils/api_utils.py:264]: They might be required later

File: src/api/constants.py [hunk 5]
```diff
@@ -37,3 +37,9 @@
 
 COMMUNITY_NOTIFICATION_TYPES = [USER_EVENTS_NUDGES_FF]
 
+PAGE_SETUP_ESSENTIALS = {
+    "homepage":{
+    
+    }
+}
+
```
Comment (Opoku-Agyemang) [inline src/api/constants.py:45]: do we need this?

File: src/api/handlers/misc.py [hunk 6]
```diff
@@ -157,3 +161,50 @@ def authenticateFrontendInTestMode(self, request):
...
+        args, err = self.validator.verify(args, strict=True)
+        if err:
+            return err
+        
+        endpoints = args.pop("data", [])
+    
+        def fetch_data(endpoint):
+            endpoint = request.build_absolute_uri(endpoint)
+            response = requests.post(endpoint,data=args)
+            return response.json()
```
Comment (Opoku-Agyemang) [inline src/api/handlers/misc.py:184]: if we do this, we lose the cookies of the user right?  can we make sure the cookies of the user are passed through?  so in essence, we assume their role and make those requests on their behalf?

File: src/api/services/misc.py [hunk 7]
```diff
@@ -141,3 +141,20 @@ def authenticateFrontendInTestMode(self, args):
...
         client = Client()
         return signinAs(client, user), None
+    
+    
+    # def load_essential_initial_site_data(self,context, args):
+    #     res, err = self.store.load_essential_initial_site_data(context,args)
+    #     if err:
+    #         return None, err
+    #
+    #     return res, None
```
Comment (Opoku-Agyemang) [inline src/api/services/misc.py:151]: remove if not needed

File: src/api/store/misc.py [hunk 8]
```diff
@@ -492,3 +477,83 @@ def list_commonly_used_icons(self):
...
+    #         community, _ = get_community(community_id=community_id, subdomain=subdonain)
+    #         if not community:
+    #             return None, CustomMassenergizeError("Community not found")
+    #
+    #         page_settings = {
+    #             "home_page_settings": HomePageSettings.objects.filter(community=community).first(),
+    #             "actions_page_settings": ActionsPageSettings.objects.filter(community=community).first(),
+    #             "events_page_settings": EventsPageSettings.objects.filter(community=community).first(),
+    #             "vendors_page_settings": VendorsPageSettings.objects.filter(community=community).first(),
+    #             "about_us_page_settings": AboutUsPageSettings.objects.filter(community=community).first(),
```
Comment (Opoku-Agyemang) [inline src/api/store/misc.py:504]: remove

File: src/api/utils/api_utils.py [hunk 9]
```diff
@@ -98,3 +108,283 @@ def create_media_file(file, name):
...
+            events = events.filter(Q(user__id=context.user_id) | Q(is_published=True))
+        else:
+            events = events.filter(is_published=True)
+    all_events = [*events, *shared]
+    
+    return all_events
+
+
+def get_teams_for_community(community):
+    pass
```
Comment (Opoku-Agyemang) [inline src/api/utils/api_utils.py:234]: why do we need this?

File: src/api/utils/api_utils.py (hunk 4 above)
Comment (Opoku-Agyemang) [reply src/api/utils/api_utils.py:264]: remove and put in a different file or branch so that we don't clutter this PR

**Review by Opoku-Agyemang — CHANGES_REQUESTED** [review]

File: src/api/handlers/misc.py (hunk 6 above)
Comment (abdullai-t) [reply src/api/handlers/misc.py:184]: I'll conduct tests with authenticated users to verify whether the cookie is being passed. If it's not, I'll ensure that it's included in the request

File: src/api/utils/api_utils.py [hunk 10]
```diff
@@ -98,3 +108,283 @@ def create_media_file(file, name):
     media = Media.objects.create(name=name, file=file)
     media.save()
     return media
+
+
+class DonationPageSettings:
```
[the remark this answers is not in the document]
Comment (abdullai-t) [reply src/api/utils/api_utils.py:113]: no

File: src/api/utils/api_utils.py (hunk 4 above)
Comment (abdullai-t) [reply src/api/utils/api_utils.py]: ok

File: src/api/utils/api_utils.py (hunk 3 above)
Comment (abdullai-t) [reply src/api/utils/api_utils.py:151]: It's not currently in use, but I've added it in case we choose to implement it in the future.

Comment (BradHN1) [reply src/api/utils/api_utils.py:151]: It is in use, but it is DonatePageSettings, not DonationPageSettings

**Pull request closed without merging**